src/MorphMapGenerator.php

```diff
@@ -39,7 +46,9 @@ private function resolveMorphFromModelClass(ReflectionClass $reflection): ?array
         $model = $reflection->newInstanceWithoutConstructor();
 
         try {
-            $morph = $model->getMorphClass();
+            $morph = static::$resolveCallback
+                ? call_user_func(static::$resolveCallback, $model)
```

Comment:
I think it would be nice to support having the `resolveCallback` returning `null`. When `null` is returned, we could fallback to `getMorphClass`